### ISSUE TITLE
Problem with bad document

### ISSUE THREAD

**Issue by SusanBrown** [cwrc/CWRC-GitWriter#23]
User lwood03 created a template document using CWRC-Writer, based on the DToC template in the cwrc documentation a week ago, swapping out the processing instructions in the template for the ones in the blank document template in github.

Today she opened it and pasted (yes!) some text into it, and when she validated got the error:
"An error occurred while trying to validate the document."

It never finishes. If she clears the error message then it continues to try to validate until she clicks "Clear" button.

[In replicating the result a second time, after this point she got the same error message even while **not** validating.]

We speculate that the text she pasted in was not clean and that the document is no longer well formed XML. 

After this point, trying to save produced the message "Error saving document".  And then the document was no longer in her list of CWRC-Writer enabled documents for her github repo. The repo is still there but CWRC-Writer no longer registers it. 

Document in question is: https://github.com/lwood03/DTOC-Template/blob/master/document.xml

Was on a Mac using Chrome.
____

I would like to discuss how we might provide better error feedback.

**Comment by ajmacdonald** [comment]
I agree that better error feedback is needed.
In this particular case I don't have enough info to determine what went wrong. I just tried pasting the XML from the linked document into the edit source dialog and was able to successfully validate it afterwards.

**Comment by ilovan** [comment]
Started a troubleshooting documentation file  for GitWriter - will include instructions on how to handle deletion of the encoding description in the header.

**Comment by jchartrand** [comment]
One of the comments above is that she can't see the document in her GitWriter enabled docs anymore, but I can see the document from the GitWriter, while logged in as myself:

![image](https://user-images.githubusercontent.com/547165/37978659-831ecc76-31b4-11e8-9418-bc5753f2106a.png)

and it appears to load fine into the GitWriter.

However, I think you'd also said during today's call that she'd created multiple versions of the document and that possibly just one of them couldn't be seen.  
My main point here is just that the application tag (which is in her document) seems to be doing it's job, correctly filtering CWRC-enabled repos.

**Comment by SusanBrown** [comment]
It seems to me that somehow whatever the problem was was somehow rectified in the interim between now and then, which makes it very hard to troubleshoot. I was in the room and saw that there was nothing in her documents tab.


On Mar 27, 2018, at 11:50 AM, James Chartrand <notifications@github.com<mailto:notifications@github.com>> wrote:


One of the comments above is that she can't see the document in her GitWriter enabled docs anymore, but I can see the document from the GitWriter, while logged in as myself:

[image]<https://user-images.githubusercontent.com/547165/37978659-831ecc76-31b4-11e8-9418-bc5753f2106a.png>

and it appears to load fine into the GitWriter.

However, I think you'd also said during today's call that she'd created multiple versions of the document and that possibly just one of them couldn't be seen.
My main point here is just that the application tag (which is in her document) seems to be doing it's job, correctly filtering CWRC-enabled repos.

**Comment by jchartrand** [comment]
Right, and a best guess is that the ‘application’ tag may have been taken out at the point when the document wasn’t showing up in the documents tab, but was then subsequently put back in, and now the document again shows in the documents tab.
